src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
```

Comment:
For `ngx.exit(ngx.ERROR)` and `ngx.exit(444)`, it just makes no sense to send out a last buf or returning `NGX_OK` to the output body filter caller.

Reply:
OK, we only allow 444 here now. https://github.com/openresty/lua-nginx-module/pull/1009/files#diff-198bce3a7eb5e4a6fdc54aae29b46483R381